Allow changing the Houdini generated output directory

endigma opened the pull request:
Fixes #391 

# Configurable output directory

This PR adds a config value to configure the output directory name for Houdini's generated files (`$houdini`) by default.

## TODO

- [x] fixup `cmd/init` to work properly
- [x] somehow drag config into `is_root_route` in `houdini-svelte/fsPatch`
- [x] update `e2e` to use `.houdini`
- [x] changset?
- [x] docs update

SeppahBaws commented:
uh-oh, can you add `.houdini/` to the gitignore? 😅

endigma commented:
whoops

endigma commented:
Running `pnpm test` on this branch and main results in the same 2 tests failing, and E2E is failing in GHA on build but this also succeeds locally, any insight?

endigma commented:
https://github.com/sveltejs/kit/issues/13002 blocks fixing the failing thing _properly_ but I can just switch the e2e to use the barrel exports. Let me know what you want done re: getting this done, if you want to merge the functionality ASAP I see a few options:

- Disable the client option and merge the plumbing with `$houdini` as the default
- Merge as-is, making `.houdini` the new default; add a note to docs about using the barreled exports.
- Merge as-is, but leaving `$houdini` as the default for new projects
- Wait until the SvelteKit bug is addressed to proceed (likely gonna hit a huge conflict when it comes time to resurrect unless I go and fix this upstream myself ASAP)

SeppahBaws commented on packages/houdini-svelte/src/plugin/fsPatch.ts:
```diff
@@ -280,7 +280,7 @@ function is_root_route(filepath: PathLike): boolean {
 		// ignore the src/routes that exists in the type roots
 		!filepath.includes('.svelte-kit') &&
-		!filepath.includes('$houdini')
+		!filepath.includes(_config.runtimeDir ?? '$houdini')
```
Shouldn't this default to '.houdini' if the runtimeDir hasn't been set?

endigma replied:
The default for `init` and some other things is now `.houdini`, but the hardcoded default has to remain `$houdini` to preserve backward compatibility afaik

AlecAivazis replied:
can you add a comment somewhere to remind us that this needs to match the default value?

endigma replied:
Done

SeppahBaws commented on packages/houdini/src/cmd/init.ts:
Do the TODOs in this file still apply, or can they be removed?

SeppahBaws reviewed (commented):
We should also update the documentation site to reflect the updated config structure.

endigma commented:
Tests should pass now that I've fixed the path resolving bug, and I've updated docs to the best of my knowledge, please let me know if I've done it wrong or if that file is somehow autogenerated.
